src/tests/Loader/classloader/DefaultInterfaceMethods/devirttest/devirttest.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+interface I<T>
+{
+    string DefaultTypeOf() => typeof(T).Name;
+}
+
+class Dummy { }
+
+class C : I<string>, I<object>, I<Dummy>
+{
+    string I<Dummy>.DefaultTypeOf() => "C.Dummy";
+}
+
+public static class Program
+{
+    [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+    static int Main()
```

Comment:
We should find a way to test this without AggressiveOptimization. Currently that will disable the Crossgen compiler and so we won't test the behavior in ahead of time compile scenarios.